joss-paper/paper.md

```diff
@@ -0,0 +1,130 @@
+
+- _data preprocessing_: functions for importing data from a variety of instruments and file types
+- _data validation_: easy-to-use methods for checking measurement validity
+- _model fitting_: a simple and powerful interface for quickly fitting models to analyze data
+- _model selection_: parameter error estimates and model confidence bounds
+- _visualization_: interactive and publication-ready Nyquist and Bode plots
+
+# Background
+
+Electrochemical impedance spectroscopy (EIS) is a powerful technique for noninvasively probing the physicochemical processes governing complex electrochemical systems by measuring the relationship between voltage and current (i.e. the impedance) as a function of frequency [@orazem_electrochemical_2008]. Analysis of EIS spectra often involves the construction and fitting of models to represent the system response. Equivalent circuit models, a popular method for interpreting impedance spectra, represent processes in an electrochemical system with elements in a circuit allowing quantitative characterization of the data. EIS inherently requires the system response is linear, causal and stable over the measurement time. As such, researchers must also ensure their data complies with these assumptions – often through some implementation of Kramers-Kronig analysis. Following the approach outlined above, workers have applied EIS to problems ranging from  measuring the dielectric properties of biological systems to corrosion rates of coated metals. More recently, EIS has been widely adopted for studying processes governing batteries and fuel cells.
+
+# Statement of Need
+To date, typical impedance analysis solutions have relied on either instrument-specific, proprietary software or ad hoc, lab-specific code written for internal use. In addition to making reproducible analysis difficult, these solutions can be restrictive to defining custom circuit elements and processing large datasets. By providing an open-source, community-driven package for the full impedance analysis pipeline from data management to parameter extraction and publication ready figures, `impedance.py` seeks to encourage reproducible, easy-to-use, and transparent analysis. In addition to decades of electrochemical research, many new methods for validating [@schonleber_method_2014] and analyzing [@murbach_analysis_2018; @buteau_analysis_2019] impedance spectra have been developed by researchers. By lowering the barrier to use tried-and-true methods along side cutting-edge analytical techniques via a consistent interface, `impedance.py` also serves to grow as a community repository of best-practices while facilitating the adoption of new techniques.
+
+# Current `impedance.py` functionality
+
+## Data Ingestion
```

Comment:
minor formatting: we should decide if we want to capitalize both words or only the first one

Reply:
I vote we go with only the first word capitalized. I'll also change this to `## Data preprocessing` to be consistent with the workflow outlined above.